WMF Framework/WidgetController.swift

```diff
@@ -54,18 +54,21 @@ public final class WidgetController: NSObject {
                 }
                 self.updateFeedContent(in: dataStore) {
                     // if there's no cached content group after update, return nil
-                    self.fetchCachedWidgetContentGroup(with: kind, isAnyLanguageAllowed: isAnyLanguageAllowed, in: dataStore, completion: completion)
+                    self.fetchCachedWidgetContentGroup(with: kind, isAnyLanguageAllowed: isAnyLanguageAllowed, shouldApplyPreferredLanguageVariant: shouldApplyPreferredLanguageVariant, in: dataStore, completion: completion)
                 }
                 return
             }
             completion(todaysContentGroup)
         }
     }
     
-    private func fetchCachedWidgetContentGroup(with kind: WMFContentGroupKind, isAnyLanguageAllowed: Bool, in dataStore: MWKDataStore, completion:  @escaping (WMFContentGroup?) -> Void) {
+    private func fetchCachedWidgetContentGroup(with kind: WMFContentGroupKind, isAnyLanguageAllowed: Bool, shouldApplyPreferredLanguageVariant: Bool = false, in dataStore: MWKDataStore, completion:  @escaping (WMFContentGroup?) -> Void) {
         assert(Thread.isMainThread, "Cached widget content group must be fetched from the main queue")
         let moc = dataStore.viewContext
-        let siteURL = isAnyLanguageAllowed ? dataStore.primarySiteURL : nil
+        var siteURL = isAnyLanguageAllowed ? dataStore.primarySiteURL : nil
+        if shouldApplyPreferredLanguageVariant {
```

Comment:
I pulled out the variant setting stuff here altogether, because it looks like it may be unneeded. Also, the Top read and On this day widget don't appear to trigger the path where `siteURL` is populated here.